devlib/utils/android.py

```diff
@@ -177,7 +177,12 @@ def newline_separator(self):
     # versions of the ls tool in Android pre-v7.
     def _setup_ls(self):
         command = "shell '(ls -1); echo \"\n$?\"'"
-        output = adb_command(self.device, command, timeout=self.timeout)
+        try:
+            output = adb_command(self.device, command, timeout=self.timeout)
+        except subprocess.CalledProcessError as e:
+            raise HostError(
```

Comment:
OK. Did think about this but was unsure. `adb_get_device` raises a `HostError` if it can't communicate with a device. I guess that's because the implication is that there *is no* device in that case?